File: Medium/Game with String/game-with-string.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.util.*;

class GFG{
    public static void main(String args[])throws IOException
    {
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(in.readLine());
        while(t-- > 0){
            String s = in.readLine();
            int k = Integer.parseInt(in.readLine());
            
            Solution ob = new Solution();
            System.out.println(ob.minValue(s, k));
        }
    }
}
// } Driver Code Ends


// User function Template for Java

class Solution{
    static int minValue(String s, int k){
        int[] counts = new int[26];
        s.chars().forEach(x -> ++counts[x - 'a']);
        
        PriorityQueue<Integer> pq = new PriorityQueue<Integer>((a, b) -> b - a);
        for(int c: counts) {
            pq.offer(c);
        }
        while(k-->0) {
            int count = pq.poll();
            pq.offer(count - 1);
        }
        
        return pq.stream().mapToInt(x->x*x).sum();
    }
}

File: Medium/Game with String/game-with-string.cpp
//{ Driver Code Starts
//Initial Template for C++

#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
//User function Template for C++

class Solution{
public:
    int minValue(const std::string &s, int k){
        std::array<uint32_t, 26> counts;
        counts.fill(0);
        
        for(const char c: s) {
            ++counts[c - 'a'];        
        }
        
        std::priority_queue<uint32_t> pq(counts.begin(), counts.end());
        while(k-->0) {
            uint32_t count = pq.top();
            pq.pop();
            pq.push(count - 1);
        }
        
        int32_t res = 0;
        
        while(!pq.empty()) {
            uint32_t ele = pq.top();
            pq.pop();
            res += ele * ele;
        }
        
        return res;
    }
};

//{ Driver Code Starts.

int main(){
    int t;
    cin>>t;
    while(t--){
        string s;
        int k;
        cin>>s>>k;
        
        Solution ob;
        cout<<ob.minValue(s, k)<<"\n";
    }
    return 0;
}
// } Driver Code Ends